src/components/application_manager/src/message_helper/message_helper.cc

```diff
@@ -1590,13 +1594,15 @@ MessageHelper::CreateAddVRCommandRequestFromChoiceToHMI(
       smart_objects::SmartObject msg_params =
           smart_objects::SmartObject(smart_objects::SmartType_Map);
       msg_params[strings::app_id] = app->app_id();
-      msg_params[strings::cmd_id] =
-          (*(it->second))[strings::choice_set][j][strings::choice_id];
-      msg_params[strings::vr_commands] =
-          (*(it->second))[strings::choice_set][j][strings::vr_commands];
       msg_params[strings::type] = hmi_apis::Common_VRCommandType::Choice;
       msg_params[strings::grammar_id] = choice_grammar_id;
-
+      msg_params[strings::cmd_id] =
+          (*(it->second))[strings::choice_set][j][strings::choice_id];
+      if ((*(it->second))[strings::choice_set][j].keyExists(
+              strings::vr_commands)) {
+        msg_params[strings::vr_commands] =
+            (*(it->second))[strings::choice_set][j][strings::vr_commands];
```

Comment:
```suggestion
      if (!(*(it->second))[strings::choice_set][j].keyExists(
              strings::vr_commands)) {
        continue;
```
Seems like we should just skip over this request if it doesn't have any VR commands. Should reduce the logic in this PR quite a bit